src/printer.cpp

```diff
@@ -77,48 +77,49 @@ void Printer::printJson(const std::vector<InventoryItem>& items) const
             continue;
         }
 
-        json_object* jsonItem = json_object_new_object();
+        nlohmann::json jsonItem = nlohmann::json::object();
 
         // print properties
         for (const auto& property : item.properties)
         {
-            json_object* jsonProp = nullptr;
+            nlohmann::json jsonProp;
 
             // get value from variant
             std::visit(
                 [&jsonProp](auto&& arg) {
                     using T = std::decay_t<decltype(arg)>;
                     if constexpr (std::is_same_v<T, bool>)
-                        jsonProp = json_object_new_boolean(arg);
+                    {
+                        jsonProp = arg;
```

Comment:
Now we don't need to have "if constexpr".

Reply:
An empty string is not an empty JSON object. ;)
I've moved the empty string checks outside of visitor.